src/transaction.js

```diff
@@ -196,61 +300,38 @@ Transaction.fromBuffer = function (buffer, network = networks.bitcoin, __noStric
     tx.expiryHeight = readUInt32()
   }
 
+  if (coins.isZcash(network) && tx.version >= Transaction.ZCASH_SAPLING_VERSION) {
+    tx.valueBalance = readUInt64()
+  }
+
   if (coins.isZcash(network) && tx.version >= Transaction.ZCASH_JOINSPLITS_SUPPORT_VERSION) {
```

Comment:
I think the second half of this if statement (`tx.version >= Transaction.ZCASH_JOINSPLITS_SUPPORT_VERSION`) should be moved (see comment below).

Reply:
agreed, I had some trouble looking at the spec and figuring out what versions were which. It'd be easier to read if the if statements were simpler (like Paddy proposed)